dali/pipeline/operator/builtin/input_operator.cc

```diff
@@ -115,4 +115,60 @@ InputOperator<GPUBackend>::ForwardCurrentData(TensorList<GPUBackend> &target, cu
   }
 }
 
+
+template<>
+void InputOperator<MixedBackend>::ForwardCurrentData(
+        TensorList<GPUBackend> &target, cudaStream_t stream) {
+  std::list<uptr_tl_type> tensor_list_elm;
+  InputSourceState state_info;
+  {
+    std::unique_lock<std::mutex> busy_lock(busy_m_);
+    tensor_list_elm = tl_data_.PopFront();
+    state_info = state_.front();
+    state_.pop_front();
+  }
+  target.Copy(*tensor_list_elm.front(), stream);
+
+  tensor_list_elm.front()->set_order(internal_copy_order_);
+
+  RecycleBuffer(tensor_list_elm);
+}
+
+
+template<>
+void InputOperator<MixedBackend>::ForwardCurrentData(
+        TensorList<CPUBackend> &target, ThreadPool &thread_pool) {
+  std::list<uptr_tl_type> tensor_list_elm;
+  {
+    std::unique_lock<std::mutex> busy_lock(busy_m_);
+    tensor_list_elm = tl_data_.PopFront();
+    state_.pop_front();
+  }
+  // if the output is pinned and input not it needs to be copied
+  if (target.is_pinned() && !tensor_list_elm.front()->is_pinned()) {
+    const auto &shapes = tensor_list_elm.front()->shape();
```

Comment:
nitpick: `auto &elem = tensor_list_elm.front();` (you use it several times in this scope)